### PR TITLE
Add code-spell action

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

**How does this PR accomplish the above?:**

**What documentation changes (if any) are needed to support this PR?:**

### CODE REVIEW

**Review by yubiuser — APPROVED** [review]

File: .codespellignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+doubleclick
+thur
+neet
+bre
+wan
```
Comment (peternewman) [inline .codespellignore:5]: I've submitted a PR which will remove some of these in due course:
https://github.com/codespell-project/codespell/pull/2337

File: .codespellignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+keep-alives
```
Comment (peternewman) [inline .codespellignore:6]: I kind of agree with this one, but I'm kind of torn as the normal spell checkers often flag it as a typo...

File: .github/workflows/codespell.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: Codespell
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  spell-check:
+    if: github.event.pull_request.draft == false
```
Comment (peternewman) [inline .github/workflows/codespell.yml:8]: ❤️ 
That's cool, although does mean someone couldn't find typos in their PR until it's open for review...

File: .github/workflows/codespell.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    -
+      name: Checkout repository
+      uses: actions/checkout@v3
+    -
+      name: Spell-Checking
+      uses: codespell-project/actions-codespell@master
+      with:
+        ignore_words_file: .codespellignore
```
Comment (peternewman) [inline .github/workflows/codespell.yml:18]: FWIW you can tuck this away elsewhere if you want.

**Review by peternewman — COMMENTED** [review]
Just a few things for info

**Pull request merged**